components/brave_wallet_ui/components/desktop/views/portfolio/portfolio-overview.tsx

```diff
@@ -119,13 +121,22 @@ export const PortfolioOverview = () => {
     networkList
   ])
 
+  // Filters visibleTokensForSupportedChains if a selectedAccountFilter is selected.
+  const visibleTokensForFilteredAccount: BraveWallet.BlockchainToken[] = React.useMemo(() => {
+    return selectedAccountFilter.id === AllAccountsOption.id
+      ? visibleTokensForSupportedChains
+      : visibleTokensForSupportedChains.filter((token) => token.coin === selectedAccountFilter.coin)
+  }, [visibleTokensForSupportedChains, selectedAccountFilter])
+
   // This looks at the users asset list and returns the full balance for each asset
   const userAssetList: UserAssetInfoType[] = React.useMemo(() => {
-    return visibleTokensForSupportedChains.map((asset) => ({
+    return visibleTokensForFilteredAccount.map((asset) => ({
       asset: asset,
-      assetBalance: fullAssetBalance(asset)
+      assetBalance: selectedAccountFilter.id === AllAccountsOption.id
+        ? fullAssetBalance(asset)
+        : getBalance(networkList, selectedAccountFilter, asset)
     }))
-  }, [visibleTokensForSupportedChains, fullAssetBalance])
+  }, [visibleTokensForFilteredAccount, selectedAccountFilter, networkList, fullAssetBalance, getBalance])
```

Comment:
getBalance is imported and does not need to be in the decency array

Reply:
Removed!